src/hotspot/share/classfile/stringTable.cpp

```diff
@@ -806,7 +806,9 @@ oop StringTable::lookup_shared(const jchar* name, int len) {
 // This should be called when we know no more strings will be added (which will be easy
 // to guarantee because CDS runs with a single Java thread. See JDK-8253495.)
 void StringTable::allocate_shared_strings_array(TRAPS) {
-  assert(CDSConfig::is_dumping_heap(), "must be");
+  if (!CDSConfig::is_dumping_heap()) {
```

Comment:
This seems to be redundant considering that `allocate_shared_strings_array()` is only called by `preload_and_dump_impl`. Unless you expect this method to be called elsewhere eventually?